src/styles/main.scss

```diff
@@ -0,0 +1,63 @@
+@import 'variable';
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: $calendar-width;
+  gap: 1px;
+
+  &__day {
+    position: relative;
+    display: flex;
+    width: $day-size;
+    height: $day-size;
+    border: $border;
+    justify-content: center;
+    align-items: center;
+    transition-duration: 0.5s;
+
+    &:hover {
+      transform: translateY(-20px);
+      cursor: pointer;
+      background-color: #ffbfcb;
+    }
+  }
+}
+
+@for $i from 1 through 31 {
+  .calendar__day:nth-child(#{$i})::before {
+    content: '#{$i}';
+    font-size: 30px;
+    font-family: Arial, Helvetica, sans-serif;
+  }
+}
+
+@each $i, $margin in $days {
+  .calendar--start-day-#{$i} {
+    .calendar__day:first-child {
+      margin-left: ($day-size + $gap) * $margin;
```

Comment:
Ensure that the variable `$gap` is defined in `variable.scss`, as it is used in the calculation for `margin-left` in the `.calendar--start-day-#{$i}` class. This is necessary for the SCSS to compile correctly.